### DISCUSSION TITLE
Using a pregenerated MATLAB RG and using SIONNA channel models for processing and validation and a replacement for MATLAB

### DISCUSSION THREAD

**Discussion by Infiltra** [NVlabs/sionna#39] [Q&A] [answered]
So, I've been using MATLAB for quite some time now, and I find it frustrating and annoying (though it's helpful in most cases but frustrating nonetheless). But MATLAB can give RG or IQ data as a .mat file or .bin/CSV(any file where I can dump data for my preference as I wish) respectively.

What I was thinking is with that data generated from MATLAB, I can port it into SIONNA and use its channel models for getting an RX RG for some specific signals of my interest (telecom and otherwise), currently, in my code that I am working on, I am able to import that RG .mat (I am having those actual complex numbers) file using scipy, but what I am missing is how can I map this to SIONNA RG or RG mapper and use the channel for further processing and getting an RX RG and try to validate it my pass time project where I am building some channel models in C, Any help, tips, suggestions regarding this end will be really helpful. If you guys need my current script that I am working on along with the .mat file, I can send it further in this thread.

**Comment by faycalaa** [comment, marked as answer]
If my understanding is correct, assuming a single transmitter with a single transmit antenna, what is imported from the datafile is a complex-valued tensor of shape `[batch_size, num_ofdm_symbols, fft_size]`. If this is the case, and if you want to use the channel models assuming an OFDM waveform, then you would need to cast this tensor into a TensorFlow tensor, and then make it fit the expected input shape of the [OFDMChannel](https://nvlabs.github.io/sionna/api/channel.html#sionna.channel.OFDMChannel) (or [ApplyOFDMChannel](https://nvlabs.github.io/sionna/api/channel.html#sionna.channel.ApplyOFDMChannel)) classes. The data input ``x`` is expected to be complex-valued and of shape `[batch size, num_tx, num_tx_ant, num_ofdm_symbols, fft_size]`. Therefore, you would need to add the `num_tx` and `num_tx_ant` dimensions to allow broadcasting, using, e.g., the [expand_to_rank](https://nvlabs.github.io/sionna/api/utils.html?highlight=expand_to_rank#sionna.utils.expand_to_rank) utility. This should output a tensor of shape `[batch size, 1, 1, num_ofdm_symbols, fft_size]`, that can be used as input to the classes implementing the OFDM channel model of Sionna.

If the RG imported from the datafile does not have a `batch_size` dimension, then it should also be added.
